### PR TITLE
(5.x) Merge 4.x

### PR DESCRIPTION
**Pull request by asmorkalov**
OpenCV Contrib: https://github.com/opencv/opencv_contrib/pull/3637
OpenCV Extra: https://github.com/opencv/opencv_extra/pull/1149

#24562 from savuor:fix_build_sdk
#24574 from ajithvallabai:3.4
#24704 from AsyaPronina:mean_scale_for_gapi_ov
#24710 from MaximSmolskiy:do-not-dilate-binary-images-on-first-iteration-in-findChessboardCorners
#24833 from MaximSmolskiy:compensate-edge-length-in-ChessBoardDetector-generateQuads
#24976 from Kumataro:refix23416
#24984 from eplankin:3.4
#24985 from Dhanwanth1803:hardswish
#24987 from ivashmak:bugfix_infinite_loop
#25008 from Lollyrana:Fixes/typo_in_python
#25009 from asmorkalov:as/fix_issue_24679_alt
#25016 from asmorkalov/as/android_tbb_warn_supp
#25020 from y-guyon:patch-1
#25023 from asmorkalov:as/aruco_doc_fix
#25026 from akretz:fix_qrcode_bugs
#25028 from asmorkalov:as/fisheye_solvepnp
#25030 from asmorkalov:as/fisheye_initial_f
#25035 from AleksandrPanov:fix_Barcode_detectAndDecode
#25036 from asmorkalov:as/revert_#24833

Previous "Merge 4.x": #24981

### CODE REVIEW

**Comment by asmorkalov** [comment]
cc @komakai @ivashmak @lpanaf

File: modules/3d/include/opencv2/3d.hpp [hunk 1]
```diff
@@ -2603,6 +2603,44 @@ of image, we can notice that on image a) these points are distorted.
 CV_EXPORTS_W void undistortImage(InputArray distorted, OutputArray undistorted,
     InputArray K, InputArray D, InputArray Knew = cv::noArray(), const Size& new_size = Size());
 
+/**
+@brief Finds an object pose from 3D-2D point correspondences for fisheye camera moodel.
+@param objectPoints Array of object points in the object coordinate space, Nx3 1-channel or
```
Comment (opencv-alalek) [inline modules/3d/include/opencv2/3d.hpp:2608]: Keep empty line after `@brief`.

Removal is unnecessary change against 4.x:

https://github.com/opencv/opencv/blob/12aa0fe89819c02db53677c5749771a1007f623e/modules/calib3d/include/opencv2/calib3d.hpp#L4059-L4061

File: modules/3d/include/opencv2/3d.hpp [hunk 2]
```diff
@@ -2603,6 +2603,44 @@ of image, we can notice that on image a) these points are distorted.
...
+The function interally undistorts points with @ref undistortPoints and call @ref cv::solvePnP,
+thus the input are very similar. Check there and Perspective-n-Points is described in @ref calib3d_solvePnP
+for more information.
+*/
+CV_EXPORTS_W bool solvePnP( InputArray objectPoints, InputArray imagePoints,
+                            InputArray cameraMatrix, InputArray distCoeffs,
+                            OutputArray rvec, OutputArray tvec,
+                            bool useExtrinsicGuess = false, int flags = SOLVEPNP_ITERATIVE,
+                            TermCriteria criteria = TermCriteria(TermCriteria::MAX_ITER + TermCriteria::EPS, 10, 1e-8)
+                            );
```
Comment (opencv-alalek) [inline modules/3d/include/opencv2/3d.hpp:2642]: Parenthesis are aligned on 4.x branch.
Unnecessary change during this merge.

File: modules/3d/test/test_fisheye.cpp [hunk 3]
```diff
@@ -385,4 +385,32 @@ TEST_F(fisheyeTest, jacobians)
     CV_Assert (cv::norm(x2 - xpred) < 1e-10);
 }
 
+TEST_F(fisheyeTest, solvePnP)
```
Comment (opencv-alalek) [inline modules/3d/test/test_fisheye.cpp:388]: Don't mess tests order between branches to avoid further merge conflicts.

This test comes right before `TEST_F(fisheyeTest, undistortImage)` on 4.x branch:

https://github.com/opencv/opencv/blob/12aa0fe89819c02db53677c5749771a1007f623e/modules/calib3d/test/test_fisheye.cpp#L153-L175

This corresponds to the line 132 on 5.x branch.

---

If you want to fix order, then do that on 4.x branch first (or better perform reviews of origin PRs)

**Comment by asmorkalov** [comment]
@opencv-alalek Done! Please take a look again.

**Review by opencv-alalek — APPROVED** [review]
